src/librustc_ast/mut_visit.rs

```diff
@@ -1225,7 +1226,9 @@ pub fn noop_visit_expr<T: MutVisitor>(Expr { kind, id, span, attrs }: &mut Expr,
         ExprKind::Struct(path, fields, expr) => {
             vis.visit_path(path);
             fields.flat_map_in_place(|field| vis.flat_map_field(field));
-            visit_opt(expr, |expr| vis.visit_expr(expr));
+            if let StructRest::Base(expr) = expr {
```

Comment:
Would it be better to write this as a `match` with all the cases expanded?

Reply:
This case is only ever going to apply to one variant, so I don't think there's danger of missing anything like this.